### DISCUSSION TITLE
GraphQL API: following > differentiate users and organizations

### DISCUSSION THREAD

**Discussion by ivan-kleshnin** [community/community#41015] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

Github has recently allowed users to follow organizations. This introduces a new situation for the `following` field in GraphQL API.

```gql
query {
  user(login: $login) {
    login
    following(first: 1) {
      nodes {
        login
        name				
        id
      }
    }
  }	
}
```

`following` has a type of `[User]` but some entities that `$login` follows can be `Organization`, not `User`! 
The only way to differentiate them so far (at least the one that I've found) is to rely on internal ids:

```
"id": "O_kgDOAMXv5g" <-- this is an organization id
"id": "U_kgDOAE2obw" <-- this is a user id
```

The above also requires `X-Github-Next-Global-ID` to be enabled. Old ids scheme seems opaque:

```
"id": "MDEyOk9yZ2FuaXphdGlvbjEyOTcyMDA2" -- organization
"id": "MDQ6VXNlcjUwODkzOTE=" -- user
"id": "MDQ6VXNlcjExNTI5MTQ4" -- organization
```

I think we should have a **union** type in `following` to be able to query organization- and user-specific fields:

```gql
query {
  user(login: $login) {
    login
    following(first: 1) {
      nodes {
        login
        name				
        ... on Organization {
          domains
        }
        ... on User {
          bio
        }
      }
    }
  }	
}


```

**Comment by shashank-priyadarshi** [comment]
Hi @ivan-kleshnin , is this the only way still? I am trying to fetch organizations in the following list but did not find any distinguishable field so far!

**Reply by ivan-kleshnin** [reply]
@shashank-priyadarshi yes, unfortunately. Since Microsoft bought GitHub, they don't really care about public APIs nearly as much. Different priorities now 😞
